django/contrib/auth/forms.py

```diff
@@ -81,7 +81,7 @@ def widget_attrs(self, widget):
         }
 
 
-class UserCreationForm(forms.ModelForm):
+class BaseUserCreationForm(forms.ModelForm):
```

Comment:
We should document this class, add release notes, and add tests for a subclass of `BaseUserCreationForm`.

Should we add `BaseUserCreationForm.clean_username()`** stub?
```python
class BaseUserCreationForm(forms.ModelForm): 
    ...
    def clean_username(self):
        raise NotImplementedError(...)
```

Reply:
Added test for UserCreationForm, release notes, and documentation for the new class. I might be missing out on test cases. So far I've only implemented the most obvious test for the new functionality.

Reply:
Adding this code to `BaseUserCreationForm` will result in a few of the tests erroring out. The `clean_username()` method depends on `getattr(self, "clean_%s" % name)()` in `~django/forms/forms.py` (line 452), which is used by calls to `is_valid()`. Hence every test that does `assertTrue(form.is_valid())` would produce the `NotImplementedError`.